src/retrace/retrace.py

```diff
@@ -1862,41 +1870,32 @@ def prepare_debuginfo(self, vmcore, chroot=None, kernelver=None, crash_cmd=["cra
         # Obtain the list of modules this vmcore requires
         if chroot:
             with open(os.devnull, "w") as null:
-                child = Popen(["/usr/bin/mock", "--configdir", chroot, "shell",
-                               "--", "crash -s %s %s" % (vmcore, vmlinux)],
-                              stdin=PIPE, stdout=PIPE, stderr=null, encoding='utf-8')
+                crash_normal = ["/usr/bin/mock", "--configdir", chroot, "shell",
+                                "--", "crash -s %s %s" % (vmcore, vmlinux)]
         else:
-            child = Popen(crash_cmd + ["-s", vmcore, vmlinux], stdin=PIPE, stdout=PIPE, stderr=STDOUT,
-                          encoding='utf-8')
-        try:
-            t = 3600
-            if CONFIG["ProcessCommunicateTimeout"]:
-                t = CONFIG["ProcessCommunicateTimeout"]
-            stdout = child.communicate("mod\nquit", timeout=t)[0]
-        except TimeoutExpired:
-            child.kill()
-            raise Exception("WARNING: crash 'mod' command exceeded " + str(t) + " second timeout - damaged vmcore?")
-        if child.returncode == 1 and "el5" in kernelver.release:
+            crash_normal = crash_cmd + ["-s", vmcore, vmlinux]
+        stdout, returncode = self.run_crash_cmdline(crash_normal, "mod\nquit")
+        if returncode == 1 and "el5" in kernelver.release:
             log_info("Unable to list modules but el5 detected, trying crash fixup for vmss files")
             crash_cmd.append("--machdep")
             crash_cmd.append("phys_base=0x200000")
             log_info("trying crash_cmd = " + str(crash_cmd))
-            child = Popen(crash_cmd + ["-s", vmcore, vmlinux], stdin=PIPE, stdout=PIPE, stderr=STDOUT,
-                          encoding='utf-8')
-            stdout = child.communicate("mod\nquit")[0]
+            # FIXME: mock
```

Comment:
“mock” what?

Reply:
This was a notation that the code does not work if mock is used.  It is unrelated to this patch but a note to look at this in the future.

Reply:
This was put in there for vmware vmcores which require a special commandline to load the crash tool (option "phys_base=0x200000").  We have other issues with vmware such as snapshot vmem + vmss files so it would be a good candidate for a patchset like "handle vmware memory images better".

Reply:
I just checked our production system and this code is still used on a few vmcores so I probably should not just remove it right now unfortunately, though it could be slated for removal / rework in the future (say 12 mo).

Reply:
Okay, thanks for the explanation.